pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -40,6 +40,257 @@ ORDER BY
+        timestamp >= '%d'
+        AND timestamp <= '%d'
+        AND kind = 4
+        AND msgSystem = '%s'
+    GROUP BY topic, serviceName
+)
+
+SELECT
+	topic,
+	serviceName AS service_name,
+    p99,
+    COALESCE((error_count * 100.0) / total_requests, 0) AS error_rate,
+    COALESCE(total_requests / %d, 0) AS throughput
+FROM
+    producer_latency
+`, start, end, queueType, timeRange)
+	return query
+}
+
+// S3, producer topic/service overview
+func generateProducerTopicLatencySQL(start, end int64, topic, service, queueType string) string {
+	timeRange := (end - start) / 1000000000
+	query := fmt.Sprintf(`
+WITH consumer_latency AS (
+    SELECT
+        quantile(0.99)(durationNano) / 1000000 AS p99,
+		stringTagMap['messaging.destination.partition.id'] AS partition,
+        COUNT(*) AS total_requests,
+        sumIf(1, statusCode = 2) AS error_count
+    FROM signoz_traces.distributed_signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+        AND timestamp <= '%d'
+        AND kind = 4
+		AND serviceName = '%s'
+        AND msgSystem = '%s'
+		AND stringTagMap['messaging.destination.name'] = '%s'
+    GROUP BY partition
+)
+
+SELECT
+	partition,
+    p99,
+    COALESCE((error_count * 100.0) / total_requests, 0) AS error_rate,
+    COALESCE(total_requests / %d, 0) AS throughput
+FROM
+    consumer_latency
+`, start, end, service, queueType, topic, timeRange)
+	return query
+}
+
+// S3 consumer overview
+func generateConsumerLatencySQL(start, end int64, queueType string) string {
+	timeRange := (end - start) / 1000000000
+	query := fmt.Sprintf(`
+WITH consumer_latency AS (
+    SELECT
+        serviceName,
+        stringTagMap['messaging.destination.name'] AS topic,
+        quantile(0.99)(durationNano) / 1000000 AS p99,
+        COUNT(*) AS total_requests,
+        sumIf(1, statusCode = 2) AS error_count,
+        SUM(numberTagMap['messaging.message.body.size']) AS total_bytes
+    FROM signoz_traces.distributed_signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+        AND timestamp <= '%d'
+        AND kind = 5
+        AND msgSystem = '%s'
+    GROUP BY topic, serviceName
+)
+
+SELECT
+    topic,
+    serviceName AS service_name,
+    p99,
+    COALESCE((error_count * 100.0) / total_requests, 0) AS error_rate,
+    COALESCE(total_requests / %d, 0) AS ingestion_rate,
+    COALESCE(total_bytes / %d, 0) AS byte_rate
+FROM
+    consumer_latency
+ORDER BY
+    topic;
+`, start, end, queueType, timeRange, timeRange)
+	return query
+}
+
+// S3 consumer topic/service
+func generateConsumerServiceLatencySQL(start, end int64, topic, service, queueType string) string {
+	timeRange := (end - start) / 1000000000
+	query := fmt.Sprintf(`
+WITH consumer_latency AS (
+    SELECT
+        quantile(0.99)(durationNano) / 1000000 AS p99,
+		stringTagMap['messaging.destination.partition.id'] AS partition,
+        COUNT(*) AS total_requests,
+        sumIf(1, statusCode = 2) AS error_count
+    FROM signoz_traces.distributed_signoz_index_v2
```

Comment:
database and table names should not be hardcoded

Reply:
let me address this in https://github.com/SigNoz/engineering-pod/issues/1916